### ISSUE TITLE
New datasouce: aws_codecommit_repository

### ISSUE THREAD

**Issue by benfortuna** [terraform-providers/terraform-provider-aws#4785]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently Terraform doesn't support AWS CodeCommit as a datasource. Propose to support codecommit repository to retrieve  info such as repo arn, clone url, repo name, etc.

### New or Affected Resource(s)

* aws_codecommit_repository (datasource only)

### Potential Terraform Configuration

```hcl
data "aws_codecommit_repository" "my_repo" {
  repository_name = "my_repo"
}

resource "aws_cloudwatch_event_rule" "myrepo_commit_event" {
  name = "myrepo-commit"
  event_pattern = <<EOF
{
  "source": [
    "aws.codecommit"
  ],
  "detail-type": [
    "CodeCommit Repository State Change"
  ],
  "resources": [
    "${data.aws_codecommit_repository.my_repo.arn}"
  ],
  "detail": {
    "event": [
      "referenceCreated",
      "referenceUpdated"
    ],
    "referenceType": [
      "branch"
    ],
    "referenceName": [
      "${var.branch}"
    ]
  }
}
EOF
}

resource "aws_codebuild_project" "my_project_build" {
  "artifacts" {
    type = "NO_ARTIFACTS"
  }
  "environment" {
    compute_type = "BUILD_GENERAL1_SMALL"
    image = "${var.codebuild_image}"
    type = "LINUX_CONTAINER"
  }
  name = "my-project-build"
  "source" {
    type     = "CODECOMMIT"
    location = "${data.aws_codecommit_repository.my_repo.clone_url_http}"
    buildspec = "codebuild/cloudformation/my-project-build.yml"
  }
  description  = "Build my project from my_repo"
  build_timeout = "5"
  service_role = "${data.aws_iam_role.my_build_role.arn}"
}
```

### References

* None.

**Comment by saravanan30erd** [comment]
Submitted PR: #4934

**Comment by bflad** [comment]
The new `aws_codecommit_repository` data source has been merged into master and will release with version 1.25.0 of the AWS provider, likely middle of next week.

**Comment by bflad** [comment]
This has been released in [version 1.25.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1250-june-27-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
